Update documentation for new SQLX file format.

BenBirt opened the pull request:
Still to do:
- update `publish_tables.mdx` (documentation for using Dataform Web): still need to make updates to the app before updating this
- SQLX reference file needs details on new code blocks e.g. `config { }`

lewish commented on docs/pages/guides/datasets.mdx:
````diff
@@ -74,139 +73,73 @@ Any dataset that is referenced by a query will automatically be added to that qu
 ```
 
-<div style={{ marginTop: "-40px" }} />
+To reference datasets with overridden schemas, use the fully qualified name of the dataset:
````
We probably should make a point that even when just changing the name, you need to use that new name as the ref value rather than the name of the file.

BenBirt replied:
good call, done

lewish commented on docs/components/navigation.tsx:
```diff
@@ -76,13 +70,9 @@ export default class Navigation extends React.Component<IProps> {
         </Menu>
         <h4>API reference</h4>
         <Menu className={styles.menu}>
-          {[
```
If you want to remove these builder like syntaxes for calling the JS API, I can get onboard with that too. It would simplify our documentation somewhat!

BenBirt replied:
As discussed at lunch, I've removed all the reference documentation

BenBirt commented:
I have:
- removed reference docs as discussed
- updated JS API docs to match the most recent changes (not allowing non-config properties to be set when calling `publish`)
- s/if_incremental/incremental_where/g

Still to do: update the Dataform Web `publish_tables.mdx` file once there is support in Web for the new language version.
